hub/api/pytorch.py

```diff
@@ -0,0 +1,241 @@
+import os
+import pickle
+import numpy as np
+from itertools import repeat
+from collections import defaultdict
+from hub.util.keys import get_index_map_key
+from hub.core.chunk_engine.chunker import join_chunks
+from hub.core.chunk_engine.read import read_tensor_meta
+from pathos.pools import ProcessPool
+from multiprocessing import resource_tracker
+from multiprocessing.shared_memory import SharedMemory
+from hub.util.exceptions import ModuleNotInstalledException
+
+
+# TODO make this use shared memory to make on the fly transforms faster. Currently using transform slows us down by 10x
+def apply_transform(transform, sample):
+    """Used to apply transforms to a single samples"""
+    return transform(sample) if transform else sample
+
+
+def shared_memory_clear(chunk_set):
+    """Helper function that checks if an existing SharedMemory exists for any chunk in chunk_set and clears it"""
+    for chunk_name in chunk_set:
+        try:
+            shm = SharedMemory(name=chunk_name)
+            shm.close()
+            shm.unlink()
+        except:
+            pass
+
+
+def read_chunk(chunk, key):
+    """Reads a single chunk from the dataset's storage provider and stores it in the SharedMemory"""
+    remove_shared_memory_from_resource_tracker()
+    chunk_path = os.path.join(key, "chunks", chunk)
+    chunk_bytes = _hub_storage_provider[chunk_path]
+    chunk = chunk_path.split("/")[-1]
+    shm = SharedMemory(create=True, size=len(chunk_bytes), name=chunk)
+    shm.buf[:] = chunk_bytes
+    shm.close()
+    return
+
+
+def remove_shared_memory_from_resource_tracker():
```

Comment:
anywhere in docstrings you say "helper function" you should remove that from the docstring and just prefix the method name with `_` (`_remove_shared_memory_from_resource_tracker`)